validator/sawtooth_validator/networking/handlers.py

```diff
@@ -177,13 +179,45 @@ def handle(self, connection_id, message_content):
 
 
 class PingHandler(Handler):
+    def __init__(self, network, allowed_frequency=10):
+        self._network = network
+        self._last_message = TimedCache()
+        self._allowed_frequency = allowed_frequency
 
     def handle(self, connection_id, message_content):
+
         request = PingRequest()
         request.ParseFromString(message_content)
 
         ack = NetworkAcknowledgement()
         ack.status = ack.OK
+        if self._network.get_connection_status(connection_id) == \
+                ConnectionStatus.CONNECTED:
+
+            if connection_id in self._last_message:
+                del self._last_message[connection_id]
+
+            return HandlerResult(
+                HandlerStatus.RETURN,
+                message_out=ack,
+                message_type=validator_pb2.Message.NETWORK_ACK)
+
+        if connection_id in self._last_message:
```

Comment:
So this will automatically consider them unauthed after a second ping if it arrives in to small a window? (Obviously, in the case where they haven't authed)

Reply:
Correct. This is done to keep with the them of being very strict about correct behavior during the authorization period. If you think this is too strict, I could add a counter that allows them to send pings to close together up to a certain limit and then close the connection. Thoughts?

Reply:
I think that it's fine, just wanted to double check on the intent